packages/wrangler/src/experimental-assets.ts

```diff
@@ -218,32 +219,23 @@ export const syncExperimentalAssets = async (
 	return completionJwt;
 };
 
-// modified from /pages/validate.tsx
-const walk = async (
-	dir: string,
-	manifest: AssetManifest,
-	startingDir: string = dir
-) => {
-	const files = await readdir(dir);
-
+export const buildAssetsManifest = async (dir: string) => {
```

Comment:
Also consider exporting this from `workers-shared` so it is kept near the algorithm for reading the manifest.